Fix vulnerability - golang.org/x/oauth2 v0.23.0 to v0.27.0

uriziv1 opened the pull request:
Update golang.org/x/oauth2 v0.23.0 to golang.org/x/oauth2 v0.27.0.

csiblock reviewed (approved):
please make sure you run go mod tidy

you also need to make sure you sign the commit ('-s' option in 'git commit' command)

uriziv1 commented:
The error at quay.io (before the fix)
GO-2025-3488 	 Unknown	golang.org/x/oauth2	v0.23.0	0.27.0	
COPYfile:08c0a1ca10a121fc8a908b6408cd55ab00d35218b848246e4847d878f2868267 in /manager
Description
Unexpected memory consumption during token parsing in golang.org/x/oauth2

uriziv1 commented:
Done.

```
csi-odf@uriziv-worker:uriziv1/fix-vuln-golang-org-x-oauth2:~/repos/ibm-storage-odf-operator> go get
csi-odf@uriziv-worker:uriziv1/fix-vuln-golang-org-x-oauth2:~/repos/ibm-storage-odf-operator> go mod tidy
csi-odf@uriziv-worker:uriziv1/fix-vuln-golang-org-x-oauth2:~/repos/ibm-storage-odf-operator> go mod vendor
csi-odf@uriziv-worker:uriziv1/fix-vuln-golang-org-x-oauth2:~/repos/ibm-storage-odf-operator> 
csi-odf@uriziv-worker:uriziv1/fix-vuln-golang-org-x-oauth2:~/repos/ibm-storage-odf-operator> git status
On branch uriziv1/fix-vuln-golang-org-x-oauth2
Your branch is up to date with 'origin/uriziv1/fix-vuln-golang-org-x-oauth2'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   Dockerfile
        modified:   api/v1alpha1/zz_generated.deepcopy.go
        modified:   bundle/manifests/flashsystem-csi-cr_v1_configmap.yaml
        modified:   bundle/manifests/ibm-storage-odf-operator.clusterserviceversion.yaml
        modified:   bundle/manifests/odf.ibm.com_flashsystemclusters.yaml
        modified:   bundle/metadata/annotations.yaml
        modified:   bundle/metadata/dependencies.yaml
        modified:   config/console/kustomization.yaml
        modified:   config/default/manager_config_patch.yaml
        modified:   config/manager/csi.ibm.com_v1_ibmblockcsi_cr.yaml
        modified:   config/manager/kustomization.yaml
        modified:   config/samples/csi.ibm.com_v1_ibmblockcsi_cr.yaml
        modified:   console/console.go
        modified:   hack/build-operator-bundle.sh
        modified:   hack/build-operator-catalog.sh

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        manifests/
        metadata/

no changes added to commit (use "git add" and/or "git commit -a")
csi-odf@uriziv-worker:uriziv1/fix-vuln-golang-org-x-oauth2:~/repos/ibm-storage-odf-operator> 
```

All changes in git status are related to the build process so they shouldn't be committed.

uriziv1 commented:
New image was built:
https://quay.io/repository/uriziv1/ibm-storage-odf-operator/manifest/sha256:4519c77e0de4dbc52baa6e5d736190a3118379021b76421cbe4c2b5fa50bb43f?tab=vulnerabilities

Vulnerability fixed.
